File: back_instrumentos/intrumentos/src/main/java/com/example/intrumentos/Entities/PedidoDetalle.java
package com.example.intrumentos.Entities;

import com.fasterxml.jackson.annotation.JsonBackReference; // Importa esta anotación
import jakarta.persistence.*;

@Entity
@Table(name = "pedido_detalle")
public class PedidoDetalle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "cantidad")
    private Integer cantidad;

    @ManyToOne(fetch = FetchType.EAGER) // Ajusta el fetch type según tu necesidad, EAGER o LAZY
    @JoinColumn(name = "instrumento_id")
    private Instrumento instrumento;

    @ManyToOne(fetch = FetchType.LAZY) // Generalmente LAZY para relaciones de padre a hijo
    @JoinColumn(name = "pedido_id")
    @JsonBackReference // <--- AÑADE ESTA ANOTACIÓN AQUÍ
    private Pedido pedido; // Referencia al pedido padre

    // Getters y Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getCantidad() {
        return cantidad;
    }

    public void setCantidad(Integer cantidad) {
        this.cantidad = cantidad;
    }

    public Instrumento getInstrumento() {
        return instrumento;
    }

    public void setInstrumento(Instrumento instrumento) {
        this.instrumento = instrumento;
    }

    public Pedido getPedido() {
        return pedido;
    }

    public void setPedido(Pedido pedido) {
        this.pedido = pedido;
    }
}

File: front_instrumentos/src/models/PedidoDetalle.ts
import type { Instrumento } from "./Instrumento";
import type { Pedido } from "./Pedido";

export class PedidoDetalle {
  private id?: number;
  private cantidad: number;
  private instrumento: Instrumento;
  private pedido?: Pedido;

  constructor(cantidad: number, instrumento: Instrumento, id?: number, pedido?: Pedido) {
    this.id = id;
    this.cantidad = cantidad;
    this.instrumento = instrumento;
    this.pedido = pedido;
  }

  // --- Getters ---

  public getId(): number | undefined {
    return this.id;
  }

  public getCantidad(): number {
    return this.cantidad;
  }

  public getInstrumento(): Instrumento {
    return this.instrumento;
  }

  public getPedido(): Pedido | undefined {
    return this.pedido;
  }

  // --- Setters ---

  public setId(id: number): void {
    this.id = id;
  }

  public setCantidad(cantidad: number): void {
    this.cantidad = cantidad;
  }

  public setInstrumento(instrumento: Instrumento): void {
    this.instrumento = instrumento;
  }

  public setPedido(pedido: Pedido): void {
    this.pedido = pedido;
  }
}
